### ISSUE TITLE
[FR] [UI] Image Grid chart run-params header overflows when runs have many params (regression in 3.12.0)

### ISSUE THREAD

**Issue by mrpositron** [mlflow/mlflow#23512]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement
- [x] I have read and agree to submit bug reports in accordance with the issues policy

### Where did you encounter this bug?
Local machine

### MLflow version
- Client: 3.12.0
- Tracking server: 3.12.0

### System information
- OS: macOS (Darwin 25.3.0)
- Browser: Chrome
- Python: 3.x

### Describe the problem
PR #22118 (shipped in 3.12.0) added a header above each image in the Image Grid chart that lists the run's parameters (intent: *"show run params (rank, optimizer, lr) in column headers so you can tell configs apart"* — see linked FR #22117). This works well for the curated 2–4 param case the FR describes, but degrades badly when the run has many params.

In my case the run is an object-detection training job whose hyperparameter dict contains an 80-class name mapping. MLflow's normal `log_hyperparams` flattening turns this into ~100 individual params (`data/names/0=person`, `data/names/1=bicycle`, ...). The Image Grid header then renders the entire flattened list concatenated as `key=value, key=value, ...` above every image cell, taking up far more vertical space than the image itself and effectively burying the visualization.

Pinning `mlflow<3.12` (e.g. 3.11.1) removes the header and the chart is usable again, confirming this is the regression point.

### Steps to reproduce the bug
1. Start MLflow 3.12.0+ tracking server.
2. Run the script in the next section.
3. Open the run in the MLflow UI → Charts → Image Grid for `val_images/predictions`.
4. Observe that each image cell has a ~100-line header listing every flattened param, dwarfing the image.

### Code to generate data required to reproduce the bug
```python
import mlflow
from PIL import Image

mlflow.set_tracking_uri("http://localhost:5000")
mlflow.set_experiment("image-header-repro")

with mlflow.start_run():
    # Simulate a flattened many-key hyperparams dict (e.g. a class-name mapping).
    for i in range(80):
        mlflow.log_param(f"data/names/{i}", f"class_{i}")
    # A handful of "real" hyperparams on top.
    for k, v in {"lr": 0.1, "batch_size": 4, "model": "picodet/l-640", "steps": 100}.items():
        mlflow.log_param(k, v)

    img = Image.new("RGB", (640, 480), (128, 128, 128))
    for step in range(5):
        mlflow.log_image(img, key="val_images/predictions", step=step)
```

### Is the console panel in DevTools showing errors relevant to the bug?
N/A

### Does the network panel in DevTools contain failed requests relevant to the bug?
N/A

### Suggested mitigations
- Truncate the header after N params or N characters with a "show more" affordance.
- Allow per-chart selection of which params appear in the header (allow/deny list).
- Add a per-chart toggle to hide the header entirely.

Related: #22117 (parent FR), #22118 (implementation).

**Comment by mrpositron** [comment]
Of the three mitigations, I'd lean toward the per-chart toggle to show/hide the run-params header. Smallest scope, default-on preserves the new behavior, and gives users with many params a clean opt-out.

**Comment by WeichenXu123** [comment]
> per-chart toggle to show/hide the run-params header

sounds good to me . do you have time to contribute the feature ? :)

**Comment by mrpositron** [comment]
Sure, sounds good to me. I’d be happy to contribute to the feature when I have some time :)

**Comment by timlod** [comment]
We're also running into this issue, but I think a per-run toggle (default on) would still not work for us.
We're logging many images per run, and right now we can't view any single one of them because the header dominates the entire view. Toggling every single image would take a long time. Could this be default-off, or toggling for all charts?
#23705
